File: tests/php/testcases/RaisesExceptionTest.php
<?
require_once 'base.php';

class RaisesExceptionTestCase extends TestServiceTestCase {

    public function test() {
        try {
            $this->service->raises_exception();
        } catch (SoapFault $e) {
            $this->assertEquals($e->getMessage(), 'hello error');
            return;
        }
        $this->assertTrue(false, 'Exception hasn\'t been thrown');
    }
}

File: tests/mono/testcases/RaisesExceptionTest.cs
using System;
using NUnit.Framework;

namespace TestCases {

    [TestFixture()]
    public class RaisesExceptionTest : TestServiceTest {

        [Test()]
        public void test_null() {
            try {
                service.raises_exception("hello");
            } catch(Exception e) {
                Assert.AreEqual("hello error", e.Message);
                return;
            }
            Assert.True(false, "Exception hasn't been thrown");
        }
    }
}
